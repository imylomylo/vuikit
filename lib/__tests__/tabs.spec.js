import { createVue, destroyVM, queryByTag, queryByTagAll, getSnapshot } from './_util'
import waitForUpdate from './_wait-for-update'

describe('Tabs', () => {
  let vm
  let snapshot
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    snapshot = getSnapshot(`
      <vk-tabs :activeTab="1" bottom center flip width="1-1">
        <vk-tab label="Tab 1">Content Tab 1</vk-tab>
        <vk-tab label="Tab 2" disabled>Content Tab 2</vk-tab>
        <vk-tab>
          <template slot="header">Header from slot</template>
        </vk-tab>
      </vk-tabs>
    `)
    expect(snapshot).toMatchSnapshot()

    snapshot = getSnapshot(`
      <vk-tabs-vertical :activeTab="1" flip width="1-1" contentWidth="1-2">
        <vk-tab>Content Tab 1</vk-tab>
      </vk-tabs-vertical>
    `)
    expect(snapshot).toMatchSnapshot()
  })

  it('triggers event change when a tab is clicked', done => {
    vm = createVue(`
      <vk-tabs activeTab="tab1">
        <vk-tab alias="tab1">Content Tab 1</vk-tab>
        <vk-tab alias="tab2">Content Tab 2</vk-tab>
      </vk-tabs>
    `)

    const cb = jest.fn()
    const tabsWrapper = queryByTag(vm, 'vk-tabs')
    const tabs = queryByTagAll(vm, 'vk-tab')

    tabsWrapper.$on('change', cb)
    waitForUpdate(() => {
      tabs[1].$el.children[0].click()
    }).then(() => {
      const args = cb.mock.calls[0]
      expect(cb).toHaveBeenCalled()
      expect(args).toHaveLength(1)
      expect(args[0]).toEqual('tab2')
    }).then(done)
  })
})
