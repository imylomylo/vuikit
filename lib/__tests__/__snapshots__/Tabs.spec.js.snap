exports[`Tabs renders correctly 1`] = `
"html
  body
    .uk-flex.uk-flex-column-reverse(center=\'\', flip=\'\', width=\'1-1\')
      ul.uk-tab.uk-tab-bottom
        li.uk-active
          a Tab 1
        |  
        li.uk-disabled
          a Tab 2
        |  
        li
          a Header from slot
      .uk-margin
        div Content Tab 1
"
`;

exports[`Tabs renders correctly 2`] = `
"html
  body
    .uk-flex.uk-flex-column-reverse(center=\'\', flip=\'\', width=\'1-1\')
      ul.uk-tab.uk-tab-bottom
        li.uk-active
          a Tab 1
        |  
        li.uk-disabled
          a Tab 2
        |  
        li
          a Header from slot
      .uk-margin
        div Content Tab 1
    .uk-grid(flip=\'\', width=\'1-1\', contentwidth=\'1-2\')
      .uk-width-auto
        ul.uk-tab.uk-tab-left
          li.uk-active
            a
      .uk-width-expand
        div
          div Content Tab 1
"
`;
